packages/hagrid/hagrid/cli.py

```diff
@@ -1388,13 +1570,39 @@ def create_launch_cmd(
     )
 
 
+def pull_command(cmd: str, kwargs: TypeDict[str, Any]) -> TypeList[str]:
+    pull_cmd = str(cmd)
+    if kwargs["release"] == "production":
+        pull_cmd += " --file docker-compose.yml"
+    else:
+        pull_cmd += " --file docker-compose.pull.yml"
```

Comment:
Q: Do we have to add headscale container also to `docker-compose.pull.yaml` for the network nodes, or maybe it is taken care at the next step.

Reply:
We need to build our own `headscale` image in the [`docker-compose.build.yml`](https://github.com/OpenMined/PySyft/blob/27772487c7bb55ae1098187fb3dcf03a664bea31/packages/grid/docker-compose.build.yml#L32) file.

Reply:
Ohh, but I realized that we also have a docker hub image as well! Yeah, we should add that.

Reply:
Just checked right now, and the docker-compose.yml added in the next steps already switch between the dev/stable version. So yup, it's taken care of at the next step.